packages/flutter/lib/src/painting/image_resolution.dart

```diff
@@ -117,6 +117,86 @@ const double _kLowDprLimit = 2.0;
 /// AssetImage('icons/heart.png')
 /// ```
 ///
+/// {@tool snippet}
+///
+/// The following shows the code required to write a widget that fully conforms
+/// to the [AssetImage] and [Widget] protocols. (It is essentially a
+/// bare-bones version of the [widgets.Image] widget made to work specifically for
+/// an [AssetImage].)
+///
+/// ```dart
+/// class MyImage extends StatefulWidget {
+///   const MyImage({
+///     Key? key,
+///     required this.assetImage,
+///   }) : super(key: key);
+///
+///   final AssetImage assetImage;
+///
+///   @override
+///   _MyImageState createState() => _MyImageState();
```

Comment:
@goderbauer Sir, the analyse-linux tests are failing because they want me to avoid using a private variable here. Should I change this? I am asking this because it has been done in the same way in the docs for Image Provider.

Reply:
This was recently changed to the following in the [ImageProvider doc](https://master-api.flutter.dev/flutter/painting/ImageProvider-class.html):

```suggestion
///   State<MyImage> createState() => _MyImageState();
```